src/Types.jl

```diff
@@ -171,18 +172,30 @@ mutable struct PackageSpec
     uuid::UUID
     version::VersionTypes
     mode::Symbol
-    PackageSpec(name::String, uuid::UUID, version::VersionTypes) =
-        new(name, uuid, version, :project)
-end
-PackageSpec(name::String, uuid::UUID) =
+    path::String
+    url::String
+    beingfreed::Bool # TODO: Get rid of this field
+    PackageSpec(name::AbstractString, uuid::UUID, version::VersionTypes, project::Symbol=:project,
+            path::AbstractString="", url::AbstractString="", beingfreed=false) =
+    new(name, uuid, version, project, path, url)
```

Comment:
This misses to initialize the `PackageSpec` meaning that `beingfreed` will contain garbage memory giving the bad result.